tests/db_functions/text/test_regexpstrindex.py

```diff
@@ -0,0 +1,135 @@
+        queryset = Author.objects.annotate(index=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 0),
+            ('J. R. R. Tolkien', 0),
+        ], transform=lambda x: (x.name, x.index), ordered=False)
+
+    def test_lookahead(self):
+        expression = RegexpStrIndex('name', Value(r'(R\. ){2}(?=Martin)'))
+        queryset = Author.objects.annotate(index=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 8),
+            ('J. R. R. Tolkien', 0),
+        ], transform=lambda x: (x.name, x.index), ordered=False)
+
+    def test_lookbehind(self):
+        expression = RegexpStrIndex('name', Value(r'(?<=George )(R\. ){2}'))
+        queryset = Author.objects.annotate(index=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 8),
+            ('J. R. R. Tolkien', 0),
+        ], transform=lambda x: (x.name, x.index), ordered=False)
+
+    def test_substitution(self):
+        expression = RegexpStrIndex('name', Value(r'(R\. )\1'))
+        queryset = Author.objects.annotate(index=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 8),
+            ('J. R. R. Tolkien', 4),
+        ], transform=lambda x: (x.name, x.index), ordered=False)
+
+    def test_expression(self):
+        expression = RegexpStrIndex(Concat(Value('Author: '), 'name'), Value(r'(R\. ){2}'))
+        queryset = Author.objects.annotate(index=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 16),
+            ('J. R. R. Tolkien', 12),
+        ], transform=lambda x: (x.name, x.index), ordered=False)
+
+    def test_update(self):
+        Author.objects.update(age=RegexpStrIndex('name', Value(r'(Martin|Tolkien)')))
+        self.assertQuerysetEqual(Author.objects.all(), [
+            14,
+            10,
+        ], transform=lambda x: x.age, ordered=False)
+
+
+class RegexpStrIndexFlagTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        Article.objects.create(
+            title='Chapter One',
+            text='First Line.\nSecond Line.\nThird Line.',
+            written=Now(),
+        )
+
+    def test_dotall_flag(self):
+        expression = RegexpStrIndex('text', Value(r'^.*$'), Value('s'))
+        article = Article.objects.annotate(result=expression).first()
+        self.assertEqual(article.result, 1)
+
+    def test_multiline_flag(self):
+        expression = RegexpStrIndex('text', Value(r'^.*\Z'), Value('m'))
+        article = Article.objects.annotate(result=expression).first()
+        self.assertEqual(article.result, 26)
+
+    def test_extended_flag(self):
+        if connection.vendor == 'oracle':
+            # Oracle doesn't support comments in extended regular expressions.
+            pattern = Value(r"""
+                ^[^ ]*
+                \ Line\.
+            """)
```

Comment:
If it's valuable to test this case we should do this for all database and move a case with comments to a separate method (skipped on Oracle).